lib/mk-ca-bundle.pl

```diff
@@ -465,15 +466,14 @@ (%)
           }
           # Example date: 200617000000Z
           # Means 2020-06-17 00:00:00Z
-          my $done = "20". $timestamp[0] . $timestamp[1]. "-".
-              $timestamp[2]. $timestamp[3]. "-".
-              $timestamp[4]. $timestamp[5]. " ".
-              $timestamp[6]. $timestamp[7]. ":".
-              $timestamp[8]. $timestamp[9]. ":".
-              $timestamp[10]. $timestamp[11];
-          my $distrustat = `date -d "$done" -u +"%s"`;
-          my $now = `date -u +"%s"`;
-          if($now > $distrustat) {
+          my $distrustat =
+            timegm($timestamp[10] . $timestamp[11], # second
+                   $timestamp[8] . $timestamp[9],   # minute
+                   $timestamp[6] . $timestamp[7],   # hour
+                   $timestamp[4] . $timestamp[5],   # day
+                   ($timestamp[2] . $timestamp[3]) - 1, # month
+                   "20" . $timestamp[0] . $timestamp[1]); # year
+          if(time >= $distrustat) {
```

Comment:
Thanks, I think we can ignore the possible time zone glitch this might have...

Reply:
Their [time](https://perldoc.perl.org/functions/time.html) is from UTC. Also I've since read [this](https://searchfox.org/mozilla-central/source/security/nss/lib/ckfw/builtins/README#65) which IMO doesn't guarantee the 200617000000Z format. I suggest a sanity check:
~~~perl
          # sanity check
          {
            local $" = '';
            if("@timestamp" !~ /^\d{12}Z$/) {
              die "Unexpected conversion"; # or do next or something here
            }
          }
~~~

Reply:
Seems reasonable. But since `time` doesn't even *guarantee* UTC maybe risking a time zone glitch is better than `die` ? Probably better to do a warning...

Reply:
I don't know what you mean by time zone glitch. It is not local time either way. If you mean the epoch could be different then yes you're right the server could have a different epoch than the client and perl time function is not guaranteed to return UTC. On Windows where the epoch is technically 1601 perl and the C runtime will use the epoch 1970.

Reply:
I mean: perl's `time()` function doesn't guarantee that the returned time is for UTC.